hatchways/team-eagle
# Backend Auth (register and login)

In this pull request, I:
- Created `/auth/register` and `/auth/login` API routes (files listed under `routes/auth`)
  - Created validations for said routes (files listed under `validation/auth`)
- Created tests for said routes
  - [You don't need to run these as they're still a work in progress. I'm still trying to figure out how to use mocks.]

#### How to Test (on Postman)
- Run server first
- To test `/auth/register`:
  1. Pick `POST` as request type
  2. Place `localhost:3001/auth/register` in URL
  3. Go to `Body` and select `x-www-form-urlencoded`
  4. Place `name`, `email`, and `password` as keys and fill in values as needed

Successful registration should return to you an instance of the user.
Unsuccessful registration should return a customized message depending on what you're missing. Try having an empty password. It should return this to you:
```JSON
{
    "password": "Password must be at least 6 characters"
}
```

- To test `/auth/login`:
  1. Pick `POST` as request type
  2. Place `localhost:3001/auth/login` in URL
  3. Go to `Body` and select `x-www-form-urlencoded`
  4. Place `email` and `password` as keys and fill in values as needed

Successful login should return to you something like this:
```JSON
{
    "success": true,
    "token": "Bearer eyJhbGciOiJIUzI1NiI ... LfHtNFaa0GgQGqERh5eZ8dO0"
}
```
An unsuccessful registration should return a customized message depending on what you're missing.

---------

I wasn't sure how to format the JSON response. Feel free to open an issue about changing it if you think we should.

Please let me know if you have any questions
Thanks,
Hakeem = )
